### PR TITLE
Expose relay version metric

### LINKED ISSUE #1747: Expose relay version in metrics
https://github.com/paritytech/devops/issues/1935#issuecomment-1375410774

We can add it as a Prometheus metric [here](https://github.com/paritytech/parity-bridges-common/blob/master/relays/utils/src/metrics/global.rs), then expose it as something like:

```
parity_bridges_common_build_info{commit="123asdzxc", version="1.2.3"}
```

### PR DESCRIPTION
**Pull request by svyatonik**
closes #1747 

I've ended up with relying on gitlab for detecting git commit, since we don't actually need to know git commit for local builds. We may use something like [that](https://github.com/paritytech/substrate/blob/master/utils/build-script-utils/src/version.rs) instead - please say if you think we need that.

### CODE REVIEW

**Comment by svyatonik** [comment]
Fair enough - I'll make changes here

File: relays/bin-substrate/src/cli/mod.rs [hunk 1]
```diff
@@ -254,17 +255,56 @@ pub struct PrometheusParams {
...
+				"Failed to parse CARGO_PKG_VERSION variable {}: {}",
+				env_relay_version,
+				e,
+			);
+
+			Version::new(0, 0, 0)
+		});
+
+		// we'll assume that we're running on GitLab, for local builds we don't care
+		let relay_commit = option_env!("CI_COMMIT_SHA").unwrap_or_else(|| {
```
[the remark this answers is not in the document]
Comment (svyatonik) [reply relays/bin-substrate/src/cli/mod.rs]: ok, done

File: relays/bin-substrate/src/cli/mod.rs [hunk 2]
```diff
@@ -254,17 +255,56 @@ pub struct PrometheusParams {
...
+
+		let env_relay_version = option_env!("CARGO_PKG_VERSION").unwrap_or_else(|| {
+			log::debug!(
+				target: "bridge",
+				"Missing CARGO_PKG_VERSION variable while compiling. Using default value",
+			);
+
+			"0.0.0"
+		});
+		let relay_version = Version::parse(env_relay_version).unwrap_or_else(|e| {
```
[the remark this answers is not in the document]
Comment (svyatonik) [reply relays/bin-substrate/src/cli/mod.rs]: It was better in terms of rust code, but I agree it wasn't perfect :) Done

File: relays/utils/src/metrics.rs [hunk 3]
```diff
@@ -99,6 +99,40 @@ impl Default for MetricsAddress {
...
+				.const_label("version", &relay_version)
+				.const_label("commit", &relay_commit),
+			)?,
+			&registry,
+		)?
+		.set(1);
+
+		log::info!(
+			target: "bridge",
+			"Exposed {} metrtic: version={} commit={}",
```
Comment (serban300) [inline relays/utils/src/metrics.rs:127]: Typo: `metric`

File: relays/utils/src/metrics.rs [hunk 4]
```diff
@@ -99,6 +99,40 @@ impl Default for MetricsAddress {
...
+	/// Creates metrics params from metrics address.
+	pub fn new(
+		address: Option<MetricsAddress>,
+		relay_version: String,
+		relay_commit: String,
+	) -> Result<Self, PrometheusError> {
+		const BUILD_INFO_METRIC: &str = "substrate_relay_build_info";
+
+		let registry = Registry::new();
+		let relay_version = format!("{relay_version}");
```
Comment (serban300) [inline relays/utils/src/metrics.rs:111]: Do we need this ?

**Review by serban300 — APPROVED** [review]

**Pull request merged**